synapse/event_auth.py

```diff
@@ -446,7 +446,9 @@ def check_redaction(
 
     if room_version_obj.event_format == EventFormatVersions.V1:
         redacter_domain = get_domain_from_id(event.event_id)
-        redactee_domain = get_domain_from_id(event.redacts)
+        redactee_domain = None
+        if event.redacts:
+            redactee_domain = get_domain_from_id(event.redacts)
```

Comment:
I think we probably need to check it's a string as well. I'd do something like:

```suggestion
        if not isinstance(event.redacts, str):
            return False
        redactee_domain = get_domain_from_id(event.redacts)
```